CMakeLists.txt

```diff
@@ -548,6 +548,12 @@ if (MSVC)
 else ()
   # HACK : because these need to come first, before any warning demotion
   string (APPEND CMAKE_CXX_FLAGS " -Wall -Wdeprecated")
+  if (is_clang OR (is_gcc AND NOT unity))
+    # gcc flags many more instances with `-Wshadow` than clang. Many incouous
+    # shadowings are flagged in unity mode. Only turn on gcc warnings in nonunity
+    # builds
+    string (APPEND CMAKE_CXX_FLAGS " -Wshadow")
+  endif()
```

Comment:
Should we be enabling c++17 with MSVC?
eg.
```
set_property (
  TARGET ${PROJECT_NAME}
  PROPERTY CXX_STANDARD 17)```

Reply:
@miguelportilla I believe this takes care of turning on c++-17 for us: https://github.com/seelabs/rippled/blob/cleanups-17/CMakeLists.txt#L482 Do I need to do something extra for msvc (I don't have a windows system to test, but CI seems to work).

Reply:
I had not seen that, thanks. MSVC 2017 is working fine with this PR, so no further change required.